### PR TITLE
Ensure Connect shows Connect My Computer only to local users

### PR DESCRIPTION
**Pull request by ravicious**
[Connect My Computer won't work in its current form for SSO users](https://github.com/gravitational/teleport/pull/27815#discussion_r1252933094). That's because the implementation depends on updating the role list of a user which isn't as straightforward for SSO users as it is for local users.

The RPC in tshd already returns an error if it detects that the user is an SSO user, however the Electron app was still showing the UI to everybody.

Checking for an SSO user is done by grabbing `GetUserType()` from a user resource we get from the cluster on calls to `GetCluster` and then returning it with `LoggedInUser`, next to ACL.

`UserTypeFromString` was based on [`ResourceDeviceEnrollStatusFromString`](https://github.com/gravitational/teleport/blob/8a1d0e23c85154b253517a6f5ed782c9dd0e1d55/api/types/device.go#L419).

### CODE REVIEW

File: web/packages/teleterm/src/services/tshd/types.ts [hunk 1]
```diff
@@ -333,5 +333,7 @@ export type CreateConnectMyComputerRoleResponse =
 export type CreateConnectMyComputerNodeTokenResponse =
   apiService.CreateConnectMyComputerNodeTokenResponse.AsObject;
 
+export const UserType = apiCluster.LoggedInUser.UserType;
```
Comment (gzdunek) [inline web/packages/teleterm/src/services/tshd/types.ts:336]: We can export value and type at the same time (we may need the type in the future).
```suggestion
export { UserType }
``` 
I did it for `export { FileTransferDirection }` above.

**Review by gzdunek — APPROVED** [review]

**Review by ryanclark — APPROVED** [review]

File: web/packages/teleterm/src/services/tshd/types.ts (hunk 1 above)
Comment (ravicious) [reply web/packages/teleterm/src/services/tshd/types.ts]: I wanted to do this but I didn't know how to reexport something that's defined in a namespace. I think I found a solution, let me know what you think.

Comment (gzdunek) [reply web/packages/teleterm/src/services/tshd/types.ts]: Ah, I didn't notice that it is defined in a namespace. Your solution is fine, I was afraid that clicking on `UserType` somewhere in the code will open the `types.ts` file, not the actual definition. But it work as it should!

File: web/packages/teleterm/src/ui/ConnectMyComputer/permissions.test.ts
[the remark this answers is not in the document]
Comment (gzdunek) [reply web/packages/teleterm/src/ui/ConnectMyComputer/permissions.test.ts:1]: Probably they don't, but I would keep them for now, until we do something better.

**Review by gzdunek — APPROVED** [review]

**Pull request merged**